src/lib/containers/FilePreview/PreviewUtils.ts

```diff
@@ -0,0 +1,170 @@
+import { FileHandle, isS3FileHandle } from '../../utils/synapseTypes'
+import { PreviewRendererType } from './PreviewRendererType'
+
+export const IMAGE_CONTENT_TYPES = [
+  'image/bmp',
+  'image/pjpeg',
+  'image/jpeg',
+  'image/jpg',
+  'image/jpe',
+  'image/gif',
+  'image/png',
+  'image/svg+xml',
+]
+const MP4_EXTENSIONS = ['.mp4', '.m4a', '.m4p', '.m4b', '.m4r', '.m4v']
+const OGG_EXTENSIONS = ['.ogv', '.ogg']
+export const VIDEO_EXTENSIONS = [...MP4_EXTENSIONS, ...OGG_EXTENSIONS, '.webm']
+export const CODE_EXTENSIONS = [
+  '.cwl',
+  '.wdl',
+  '.json',
+  '.awk',
+  '.bat',
+  '.btm',
+  '.c',
+  '.cmd',
+  '.cpp',
+  '.cxx',
+  '.def',
+  '.dlg',
+  '.dpc',
+  '.dpj',
+  '.dtd',
+  '.h',
+  '.hdl',
+  '.hpp',
+  '.hrc',
+  '.html',
+  '.hxx',
+  '.inc',
+  '.ini',
+  '.inl',
+  '.ins',
+  '.ipy',
+  '.java',
+  '.js',
+  '.jsp',
+  '.l',
+  '.lgt',
+  '.ll',
+  '.par',
+  '.pl',
+  '.py',
+  '.r',
+  '.rc',
+  '.rdb',
+  '.res',
+  '.s',
+  '.sbl',
+  '.scp',
+  '.sh',
+  '.sql',
+  '.src',
+  '.srs',
+  '.xml',
+  '.xrb',
+  '.y',
+  '.yaml',
+  '.yxx',
+]
+export const MARKDOWN_EXTENSIONS = ['.md', '.markdown', '.rmd']
+
+function getFileExtension(fileName: string): string | null {
+  const index = fileName.lastIndexOf('.')
+  if (index === -1) {
+    return null
+  }
+  return fileName.substring(index)
+}
+
+/**
+ * Given a file handle that represents a link to original data, return the type of renderer that we could use to
+ * display the file contents in the browser.
+ *
+ * @returns the PreviewRendererType that would be used to render the file contents, or NONE if the file cannot be rendered
+ */
+export function getOriginalFileHandleRenderer(
+  dataFileHandle: FileHandle,
+): PreviewRendererType {
+  // TODO: Could this be changed to CloudProviderFileHandle?
```

Comment:
I believe so if the auth check has been delegated to Synapse for all CloudProviderFileHandle types